Tiltfile

```diff
@@ -61,11 +62,89 @@ def deploy_capi():
+    COPY --from=builder /manager /manager
+    COPY .release/ /tmp/downloads/cluster-stacks/
+    """
+else:
+    tilt_dockerfile_header_cso = """
+    FROM ghcr.io/sovereigncloudstack/cso:v0.1.0-alpha.2 as builder
+
+    FROM docker.io/library/alpine:3.18.0 as tilt
+    WORKDIR /
+    COPY --from=builder /usr/local/bin/helm /usr/local/bin/helm
+    COPY --from=builder /manager /manager
+    """
+    
+
 def deploy_cso():
-    version = settings.get("cso_version")
-    cso_uri = "https://github.com/sovereignCloudStack/cluster-stack-operator/releases/download/{}/cso-infrastructure-components.yaml".format(version)
-    cmd = "curl -sSL {} | {} | kubectl apply -f -".format(cso_uri, envsubst_cmd)
-    local(cmd, quiet = True)
+    # yaml = str(kustomizesub("./hack/observability")) # build an observable kind deployment by default
+    yaml_cso = './cso-components.yaml'
+    local_resource(
+        name = "cso-components",
+        cmd = ["sh", "-ec", sed_cmd, yaml_cso, "|", envsubst_cmd],
+        labels = ["CSO"],
+    )
+
+    entrypoint = ["/manager"]
+    extra_args = settings.get("extra_args")
+    if extra_args:
+        entrypoint.extend(extra_args)
+
+    # Set up an image build for the provider. The live update configuration syncs the output from the local_resource
+    # build into the container.
+    if settings.get("local_mode"):
+        docker_build_with_restart(
+            ref = "ghcr.io/sovereigncloudstack/cso-staging",
+            context = ".",
+            dockerfile_contents = tilt_dockerfile_header_cso,
+            target = "tilt",
+            entrypoint = entrypoint,
+            live_update = [
+                sync(".release", "/tmp/downloads/cluster-stacks"),
+            ],
+            ignore = ["templates"],
+        )
+    else:
+        docker_build_with_restart(
+            ref = "ghcr.io/sovereigncloudstack/cso-staging",
+            context = ".",
+            dockerfile_contents = tilt_dockerfile_header_cso,
+            target = "tilt",
+            entrypoint = entrypoint,
+            live_update = [
+                sync("cso-components.yaml", "/cso-components.yaml"),
+            ],
+            ignore = ["templates"],
+        )
+
+    k8s_yaml(yaml_cso)
+    k8s_resource(workload = "cso-controller-manager", labels = ["CSO"])
+    k8s_resource(
+        objects = [
+            "cso-system:namespace",
+            "clusteraddons.clusterstack.x-k8s.io:customresourcedefinition",
+            "clusterstackreleases.clusterstack.x-k8s.io:customresourcedefinition",
+            "clusterstacks.clusterstack.x-k8s.io:customresourcedefinition",
+            "cso-controller-manager:serviceaccount",
+            "cso-leader-election-role:role",
+            "cso-manager-role:clusterrole",
+            "cso-leader-election-rolebinding:rolebinding",
+            "cso-manager-rolebinding:clusterrolebinding",
+            "cso-serving-cert:certificate",
+            "cso-selfsigned-issuer:issuer",
```

Comment:
Will the remote mode work when there is no `cso-cluster-stack-variables:secret` here also I do not see it in cso-components.yaml

Reply:
I think this is not required and I'll remove it in the next commit.